src/test/java/io/camunda/testing/assertions/JobAssertTest.java

```diff
@@ -0,0 +1,285 @@
+package io.camunda.testing.assertions;
+
+import static io.camunda.testing.assertions.BpmnAssertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+import static org.assertj.core.data.Offset.offset;
+
+import io.camunda.testing.extensions.ZeebeAssertions;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.camunda.community.eze.RecordStreamSource;
+import org.camunda.community.eze.ZeebeEngine;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+// TODO remove Thread.sleeps
+@ZeebeAssertions
+class JobAssertTest {
+
+  public static final String PROCESS_INSTANCE_BPMN = "looping-servicetask.bpmn";
+  public static final String PROCESS_INSTANCE_ID = "looping-servicetask";
+  public static final String ELEMENT_ID = "servicetask";
+  public static final String WRONG_VALUE = "wrong value";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+
+  // These tests are for testing assertions as well as examples for users
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasElementId() throws InterruptedException {
+      // given
+      deployProcess(PROCESS_INSTANCE_BPMN);
+      final Map<String, Object> variables = Collections.singletonMap("totalLoops", 1);
+      startProcessInstance(PROCESS_INSTANCE_ID, variables);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse =
+          client.newActivateJobsCommand().jobType("test").maxJobsToActivate(1).send().join();
```

Comment:
🔧 Dito for the job type

Reply:
💭 It might be nice to extract the activation of jobs to a separate method as all the testcases in this class will be using it.